source/MaterialXView/Viewer.cpp

```diff
@@ -1088,11 +1094,145 @@ void Viewer::drawContents()
     }
 }
 
+mx::MeshStreamPtr Viewer::createUvPositionStream(mx::MeshPtr mesh, 
+                                                 const std::string& uvStreamName, 
+                                                 unsigned int index,
+                                                 const std::string& positionStreamName)
+{
+    mx::MeshStreamPtr uvStream3D = mesh->getStream(positionStreamName);
+    if (!uvStream3D)
+    {
+        uvStream3D = mx::MeshStream::create(positionStreamName, mx::MeshStream::POSITION_ATTRIBUTE, 0);
+        mesh->addStream(uvStream3D);
+
+        mx::MeshStreamPtr uvStream2D = mesh->getStream(uvStreamName, index);
```

Comment:
Can we assume this stream is always available? Or do we need a nullptr check on uvStream2D?

Reply:
We should probably check at the top as it's useless to display when there are no uvs. WIll fix.